server/mergin/sync/public_api_controller.py

```diff
@@ -622,17 +643,23 @@ def get_paginated_projects(
         only_public,
     )
     result = projects.paginate(page, per_page).items
-    total = projects.paginate(page, per_page).total
+    # for count (done in subquery) we only need ids and then to do distinct for deduplications due to joins
+    # https://docs.sqlalchemy.org/en/20/faq/sessions.html#my-query-does-not-return-the-same-number-of-objects-as-query-count-tells-me-why
+    total = (
```

Comment:
Why is this different total then paginate total?

Reply:
Unfortunately it is how sqlalchemy implements count() api, it does count with some subselect which in case of joins have some duplicates - I had to add join() to query to sort out proper access filters. The underlying change is that before we had 1:1 project-project_access now we have 1:n with project_member.
At the end, count() >= len(all()).

From all I read about this issue, the best way to overcome is to count in deduplicated result (therefore distinct()).

Reply:
cool, I rewritten the query to use subquery in favour of join and we do not need to worry about that anymore